### ISSUE TITLE
Failure to parse log line due to app name having spaces

### ISSUE THREAD

**Issue by dtseiler** [bucardo/tail_n_mail#23]
In my tnm.config.txt:
`LOG_LINE_PREFIX: '%t [%p]: [%l-1] db=%d,user=%u,app=%a,client=%h '`

However it's catching and reporting lines like these:

```
[1] (from line 27,630)
?
2017-10-24 20:21:34 CDT [29585]: [1207-1] db=xyzfrance,user=svc_usfood,app=PostgreSQL JDBC Driver,client=xyzabcpapp37.disruptfood.com LOG:  execute S_259: UPDATE quartz_TRIGGERS SET TRIGGER_STATE = $1 WHERE TRIGGER_NAME = $2 AND TRIGGER_GROUP = $3 AND TRIGGER_STATE = $4
[2] (from line 27,631)
?
2017-10-24 20:21:34 CDT [29585]: [1208-1] db=xyzfrance,user=svc_usfood,app=PostgreSQL JDBC Driver,client=xyzabcpapp37.disruptfood.com DETAIL:  parameters: $1 = 'ACQUIRED', $2 = 'Smoke Test Email ReportTrigger', $3 = 'DEFAULT', $4 = 'WAITING'
[3] (from line 27,632)
?
2017-10-24 20:21:34 CDT [29585]: [1209-1] db=xyzfrance,user=svc_usfood,app=PostgreSQL JDBC Driver,client=xyzabcpapp37.disruptfood.com LOG:  execute S_54: INSERT INTO quartz_FIRED_TRIGGERS (ENTRY_ID, TRIGGER_NAME, TRIGGER_GROUP, IS_VOLATILE, INSTANCE_NAME, FIRED_TIME, STATE, JOB_NAME, JOB_GROUP, IS_STATEFUL, REQUESTS_RECOVERY, PRIORITY) VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
[4] (from line 27,633)
?
2017-10-24 20:21:34 CDT [29585]: [1210-1] db=xyzfrance,user=svc_usfood,app=PostgreSQL JDBC Driver,client=xyzabcpapp37.disruptfood.com DETAIL:  parameters: $1 = 'NON_CLUSTERED1508347730715', $2 = 'Smoke Test Email ReportTrigger', $3 = 'DEFAULT', $4 = 'f', $5 = 'NON_CLUSTERED', $6 = '1508894520000', $7 = 'ACQUIRED', $8 = NULL, $9 = NULL, $10 = 'f', $11 = 'f', $12 = '5'
```

Discussed with @turnstep , he believes it to be due to whitespace in the app %a value "PostgreSQL JDBC Driver" and difficulty with regex. Creating issue to track it.

**Comment by turnstep** [comment]
(this is probably the same issue as #22 )

For the record, the problem is that a log_line_prefix gets turned into a regex so that tail_n_mail knows when the actual output starts and the prefix ends. However, if something like the %a (application_name) has a space, there is now way to tell things apart. Consider a log_line_prefix of "%a %d". Does "foo bar baz" indicate an application name of "foo", a database named "bar", and baz as the log output, or does it indicate an application name of "foo bar" and a database name of "baz" (or even a database name of "bar baz" for that matter).

**Comment by turnstep** [comment]
There is a possible solution in 83453c0bd4ebc90b883ec19189980fe84b31030f. If there is something *after* the %a in the log_line_prefix, and whatever comes after it is not variable (i.e. no percent sign), then we can attempt a non-greedy slurp of everything. In the example above, the log_line_prefix could be better written as "app=%a,db=%d", in which case we can regex with /app=.*?,db=/.

**Comment by turnstep** [comment]
Fixed in version 1.31.2
